tests/test_models/phelps_deaton.py

```diff
@@ -1,65 +1,71 @@
-"""Define example model specifications."""
+"""Example specifications of the Phelps-Deaton model."""
 
 import jax.numpy as jnp
 
-RETIREMENT_AGE = 65
-N_CHOICE_GRID_POINTS = 500
-N_STATE_GRID_POINTS = 100
+# ======================================================================================
+# Numerical parameters and constants
+# ======================================================================================
 
+N_GRID_POINTS = {
+    "states": 100,
+    "choices": 500,
+}
 
-def phelps_deaton_utility_with_shock(
-    consumption,
-    working,
-    delta,
-    additive_utility_shock,
-):
-    return jnp.log(consumption) + additive_utility_shock - delta * working
+RETIREMENT_AGE = 65
 
+# ======================================================================================
+# Model functions
+# ======================================================================================
 
-def phelps_deaton_utility(consumption, working, delta):
+
+# --------------------------------------------------------------------------------------
+# Utility functions
+# --------------------------------------------------------------------------------------
+def utility(consumption, working, delta):
     return jnp.log(consumption) - delta * working
 
 
-def phelps_deaton_utility_with_filter(
+def utility_with_filter(
     consumption,
     working,
     delta,
     lagged_retirement,  # noqa: ARG001
 ):
-    return jnp.log(consumption) - delta * working
+    return utility(consumption=consumption, working=working, delta=delta)
 
 
+# --------------------------------------------------------------------------------------
+# Auxiliary variables
+# --------------------------------------------------------------------------------------
 def working(retirement):
     return 1 - retirement
 
 
-def next_wealth_with_shock(
-    wealth,
-    consumption,
-    working,
-    wage,
-    wage_shock,
-    interest_rate,
-):
-    return interest_rate * (wealth - consumption) + wage * wage_shock * working
+def wage(age):
+    return 1 + 0.1 * age
+
+
+def age(_period):
+    return _period + 18
 
 
+# --------------------------------------------------------------------------------------
+# State transitions
+# --------------------------------------------------------------------------------------
 def next_wealth(wealth, consumption, working, wage, interest_rate):
     return (1 + interest_rate) * (wealth - consumption) + wage * working
 
 
+# --------------------------------------------------------------------------------------
+# Constraints
+# --------------------------------------------------------------------------------------
 def consumption_constraint(consumption, wealth):
     return consumption <= wealth
 
 
-def wage(age):
-    return 1 + 0.1 * age
-
-
-def age(_period):
-    return _period + 18
-
-
+# --------------------------------------------------------------------------------------
+# Filters
```

Comment:
Now I finally understand what is confusing me here in terms of terminology.

Typically, when we say "filter", we mean that we *filter out* the big particles that don't pass it.

But we define the states that should *pass through* the filter. 

Not sure there is simple terminology to turn that around, but I would find it helpful.

Reply:
Currently we have:

```python
def mandatory_retirement_filter(retirement, age):
    return jnp.logical_or(retirement == 1, age < RETIREMENT_AGE)
```

Would you prefer for users to define the negation:

```python
def mandatory_retirement_filter(retirement, age):
    return jnp.logical_and(retirement == 0, age >= RETIREMENT_AGE)
```

Reply:
No, we agreed with @janosg that ultimately, this way 'round is much clearer.

It is just a matter of terminology. And maybe even just a change that I need to make in my brain.

Reply:
I guess a better term for what we are doing right now would be "mask".

Reply:
Not sure, masking to me implies that other states are generated and hidden. 

But again, might just be a switch in my head towards filtering=keeping the parts in question.